### DISCUSSION TITLE
[@types/graphql-upload] Outdated typescript definitions

### DISCUSSION THREAD

**Discussion by marcelhas** [DefinitelyTyped/DefinitelyTyped#60381] [Issues with a @types package]
# Problem
I am using `graphql-upload@^12.0.0` in a project and the most current `@types/graphql-upload@^8.0.11` has not been updated since v8. 

# Example
Since v10 `createReadStream()` can take options as arguments (See <https://github.com/jaydenseric/graphql-upload/blob/master/changelog.md#1000>), which is currently missing.
The following snippet gives a type error, but should not:

```ts
import { FileUpload } from "graphql-upload";
const file: Promise<FileUpload> = ...
const { createReadStream } = await file;
createReadStream({ encoding: "binary" })
```
See <https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/graphql-upload/index.d.ts>.

[a comment by an automated account is not shown]

**Comment by Narretz** [comment]
graphql-upload has changed the exports in v14. So any update to the types should incorporate that.

**Reply by Shinigami92** [reply]
I currently even don't know how to import it in typescript by using e.g. `// @ts-expect-error` above the import statement.
Could you give me a hint?
`import { GraphQLUpload } from 'graphql-upload'` this currently fails.

**Reply by Otard95** [reply]
The new import syntax as described in [this](https://github.com/jaydenseric/graphql-upload/issues/324) issue would be `import { GraphQLUpload } from 'graphql-upload/GraphQLUpload.js'`.

How ever this currently gives
```
error TS7016: Could not find a declaration file for module 'graphql-upload/GraphQLUpload.js'. '<project path>/node_modules/graphql-upload/GraphQLUpload.js' implicitly has an 'any' type.
  If the 'graphql-upload' package actually exposes this module, consider sending a pull request to amend 'https://github.com/DefinitelyTyped/DefinitelyType
d/tree/master/types/graphql-upload'
``` 
when compiling with `tsc`.

EDIT: [this](https://github.com/jaydenseric/graphql-upload/issues/319) might be even more helpful.
